### ISSUE TITLE
Suggestion: Add /cspeed to control player movement speed

### ISSUE THREAD

**Issue by James103** [Earthcomputer/clientcommands#155]
Currently, there is no command to change how fast the player travels. I would like if there was a /cspeed command that took a single float between 0 and 1 inclusive (but not more than 1 to prevent cheating) and multiplied your movement speed by that value. For example, `/cspeed 0.5` makes you walk half as fast, then `/cspeed 0.7` makes you walk 70% as fast.

Example use cases:
* More precise and reliable block placement and breaking in creative mode (since they are both timing-sensitive)
* Matching your speed with whatever animal's following you to reduce the separation between them
* AFK farming in survival mode that relies on speed-sensitive mechanics

Movement actions that should be affected by `/cspeed`:
* Land movement (walking, sprinting, sneaking)
* Underwater movement (swimming, sprint-swimming)
* Creative flight
* Riding a boat or horse (if possible)

Movement actions that should not be affected by `/cspeed`:
* Any movement of the player that is initiated server-side (like riding a minecart or untamed horse)
* Flying with an elytra

**Comment by Mhowser** [comment]
I think we should be able to go faster in singleplayer only.

**Comment by Earthcomputer** [comment]
I can see the usecase for creative mode, survival mode usage feels a little cheaty imo, but i guess its not that bad as long as its limited so you cant go above the vanilla move speed

**Comment by wb1016** [comment]
this wont be able to used on survival unless server allows this mod.
it will spam 'moved too fast' on server log.

**Comment by Earthcomputer** [comment]
The server allows you to move 20 blocks per second before that happens

**Comment by leumasme** [comment]
I dont believe that it would be useful to add an arbitrary limitation to this speed value.
If people want to cheat, there will always be cheats for them. I agree that clientcommands should be a Client Utility mod and not a cheat, but i do not believe that adding arbitrary limitations to the features to avoid seeming like a cheat client is the way to do this.
There are plenty legit reasons in testing mechanics and contraptions that would benefit from increasing the player speed.

**Comment by Earthcomputer** [comment]
There is no legitimate reason to allow players to increase their movement speed in survival. It gives them a clear advantage and there is no good reason to not block them from doing this.
To be clear. The plan for this feature is:
- Allow players to reduce their movement speed in survival.
- Allow players to increase or reduce their movement speed in creative.

**Comment by leumasme** [comment]
Ignore this limitation for singleplayer then.

**Comment by DragonEggBedrockBreaking** [comment]
> 
> To be clear. The plan for this feature is:
> 
>     * Allow players to reduce their movement speed in survival.
> 
>     * Allow players to increase or reduce their movement speed in creative.

Just a reminder to increase the limit when under a speed effect, in case you want a speed faster than normal but less than speed 1 (so you get speed 1 with potion/beacon and decrease with mod command) or if you want a speed between speed 1 and 2 (you you get speed 2 with potion/beacon and decrease with mod command).

**Comment by Mr-Sbev** [comment]
I tried to implement this, but I was only successful with velocity changes which are imprecise and are pretty annoying when playing. I am still learning to code, so I might have missed something.

**Comment by m-sarabi** [comment]
it will be really helpful to synchronize the building speed with movement speed while as with speed effect you go too fast that building speed can't keep up and spamming the use can place the block where you don't want it
